### ISSUE TITLE
feature(color-picker, time-picker): Provide mobile variant of picker popup

### ISSUE THREAD

**Issue by motabass** [tiaguinho/material-community-components#170]
Right now using the pickers on small mobile devices can get cut off:
![image](https://user-images.githubusercontent.com/4824274/93709339-ef9f6900-fb3d-11ea-9dd6-0427be12a106.png)

MatDatePicker solves this by using a centered popup for mobile devices

**Comment by jamesrusso** [comment]
I agree that simply centered would be the reasonable solution here and it should be possible to just use the underlying CDK overlay functionality "centerHorizontally" & "centerVertically" for certain viewport sizes maybe?

**Comment by motabass** [comment]
I did a workaround for now which prevents it from being cut-off but still does not center it:
https://github.com/tiaguinho/material-community-components/pull/178/commits/2cda8bdd0e2a22af7a5ebd6f30d7d6aa9c968a9b
